contracts/contracts/token/WrappedOusd.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { ERC20 } from "../../lib/solmate/src/tokens/ERC20.sol";
+import { ERC4626 } from "../../lib/solmate/src/mixins/ERC4626.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { OUSD } from "./OUSD.sol";
+import { Governable } from "../governance/Governable.sol";
+
+contract WrappedOusd is ERC4626, Governable {
+    using SafeERC20 for IERC20;
+
+    constructor(
+        ERC20 _underlying,
+        string memory _name,
+        string memory _symbol
+    ) ERC4626(_underlying, _name, _symbol) Governable() {
+        OUSD(address(_underlying)).rebaseOptOut(); // It's not treated as a contract yet
+        OUSD(address(_underlying)).rebaseOptIn();
+    }
+
+    /**
+     * @notice Show the total amount of OUSD held by the wrapper
+     */
+    function totalAssets() public view override returns (uint256) {
+        return ERC20(asset).balanceOf(address(this));
```

Comment:
this is possibly manipulatable in a cream-hack esque manner no? would make wOUSD unsuitable for borrowing in lending markets

Reply:
The issue is that you can flat out give OUSD to wOUSD, making each wOUSD worth more. This is normal a bad thing for an attacker. wOUSD holders, and OUSD holders both are happy and good here.

The problem is if a lending platform is using wOUSD as collateral, has lended it back to the attacker and suddenly thinks the attacker can borrow a bajillion dollars.


See these two threads:

- https://twitter.com/danielvf/status/1453426746326405130
- https://twitter.com/danielvf/status/1453460110760677379

This isn't a vulnerability in wOUSD, just something a lending platform has to take into account.